src/earn/EarnCollectScreen.tsx

```diff
@@ -40,7 +47,34 @@ export default function EarnCollectScreen({ route }: Props) {
     feeCurrencies,
   })
   const onPress = () => {
-    // TODO(ACT-1180): submit the tx
+    if (!asyncRewardsInfo.result || asyncPreparedTransactions.result?.type !== 'possible') {
```

Comment:
```suggestion
    if (!asyncRewardsInfo.result || (asyncPreparedTransactions.result && asyncPreparedTransactions.result.type !== 'possible')) {
```

nit: to get rid of optional chaining